### ISSUE TITLE
0.5.8 fails to start with panic: sync: negative WaitGroup counter

### ISSUE THREAD

**Issue by evilhamsterman** [kube-vip/kube-vip#505]
**Describe the bug**
The just released version v0.5.8 crashes on startup. I just deployed a fresh clean k3s cluster which pulled the latest version but the kube-vip-ds pods all just crash. If I revert to v0.5.7 everything works fine

**To Reproduce**
1. Deploy k3s with kube-vip daemonsets
2. Try to create a load balancer service and see that External IP is stuck <Pending>
3. View the pods in kube-system and see all the kube-vip-ds pods crashing
4. Check the logs and see
```
panic: sync: negative WaitGroup counter

goroutine 55 [running]:
sync.(*WaitGroup).Add(0x1cf14c8?, 0xc0000c6008?)
        /usr/local/go/src/sync/waitgroup.go:83 +0xda
sync.(*WaitGroup).Done(...)
        /usr/local/go/src/sync/waitgroup.go:108
github.com/kube-vip/kube-vip/pkg/manager.(*Manager).servicesWatcher(0xc000226310, {0x1cf1490?, 0xc000272d00}, 0xc0000ade20)
        /src/pkg/manager/watch_services.go:161 +0x11ef
github.com/kube-vip/kube-vip/pkg/manager.(*Manager).startARP.func3({0x1cf1490, 0xc000272d00})
        /src/pkg/manager/manager_arp.go:156 +0x86
created by k8s.io/client-go/tools/leaderelection.(*LeaderElector).Run
        /go/pkg/mod/k8s.io/client-go@v0.25.2/tools/leaderelection/leaderelection.go:211 +0x11b
```

**Expected behavior**
The pods not to crash and load balancing to work

**Screenshots**

**Environment (please complete the following information):**
 - OS/Distro: Ubuntu 22.04
 - Kubernetes Version: v1.25.5+k3s2
 - Kube-vip Version: 0.5.8

**`Kube-vip.yaml`:**
```yaml
---
apiVersion: apps/v1
kind: DaemonSet
metadata:
  labels:
    app.kubernetes.io/name: kube-vip-ds
    app.kubernetes.io/version: v0.5.8
  name: kube-vip-ds
  namespace: kube-system
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: kube-vip-ds
  template:
    metadata:
      labels:
        app.kubernetes.io/name: kube-vip-ds
        app.kubernetes.io/version: v0.5.8
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: node-role.kubernetes.io/master
                    operator: Exists
              - matchExpressions:
                  - key: node-role.kubernetes.io/control-plane
                    operator: Exists
      containers:
        - args:
            - manager
          env:
            - name: vip_arp
              value: "true"
            - name: port
              value: "6443"
            - name: vip_interface
              value: ens192
            - name: vip_cidr
              value: "32"
            - name: cp_enable
              value: "true"
            - name: cp_namespace
              value: kube-system
            - name: vip_ddns
              value: "false"
            - name: svc_enable
              value: "true"
            - name: vip_leaderelection
              value: "true"
            - name: vip_leaseduration
              value: "5"
            - name: vip_renewdeadline
              value: "3"
            - name: vip_retryperiod
              value: "1"
            - name: address
              value: 10.100.0.107
            - name: prometheus_server
              value: :2112
            - name: enableServicesElection
              value: "true"
          image: ghcr.io/kube-vip/kube-vip:v0.5.8
          imagePullPolicy: Always
          name: kube-vip
          resources: {}
          securityContext:
            capabilities:
              add:
                - NET_ADMIN
                - NET_RAW
      hostNetwork: true
      serviceAccountName: kube-vip
      tolerations:
        - effect: NoSchedule
          operator: Exists
        - effect: NoExecute
          operator: Exists
  updateStrategy: {}
status:
  currentNumberScheduled: 0
  desiredNumberScheduled: 0
  numberMisscheduled: 0
  numberReady: 0

---
apiVersion: v1
kind: ConfigMap
metadata:
  name: kubevip
  namespace: kube-system
data:
  range-global: 192.168.1.1-192.168.1.10
```

**Additional context**

**Comment by wolfnfox** [comment]
I also got caught out by this error.

What was slightly more problematic is the `ports` previously assigned were not released after the crash and so the ports remained <unset> even though the VIP was up and broadcasting.

This meant the service was unreachable with a connection was reset by the peer error.  I was forced to manually restart the server that had picked up the VIP to clear the ports and free up the VIP to go to another server.

Although I can understand the VIP not being dependant on a port being free, the fact the port was able to be allocated was still effectively in a failed state, perhaps I'm missing something but if the kube-vip cannot fulfill both the port allocation and the VIP surely it should give up its claim and allow another kube-vip service to inherit the VIP?

**Comment by thebsdbox** [comment]
Strange, I'll need to improve the release testing as some of the waiting group logic has broken the behaviour of running it as a daemonset :|

**Comment by thebsdbox** [comment]
Found the bug, that was added in error.

**Comment by thebsdbox** [comment]
This is fixed in `0.5.9`, out of sync `wait.Done()`
